src/core/Rational.pm6

```diff
@@ -94,35 +94,38 @@ my role Rational[::NuT = Int, ::DeT = ::("NuT")] does Real {
         my $whole  = self.abs.floor;
         my $fract  = self.abs - $whole;
 
+        my $reduce = False;
         # fight floating point noise issues RT#126016
-        if $fract.Num == 1e0 { ++$whole; $fract = 0 }
+        if $fract.Num == 1e0 {
+            if nqp::unbox_s(self.^name) eq 'Rat' {
+                $whole += 1;
+                $fract = 0
+            }
+            $reduce = True
+        }
 
         my $result = nqp::if(
             nqp::islt_I($!numerator, 0), '-', ''
         ) ~ $whole;
 
         if $fract {
-            my $precision = $!denominator < 100_000
-                ?? 5 !! $!denominator.Str.chars;
-
-            my $fract-result = '';
-            while $fract and $fract-result.chars < $precision {
-                $fract *= 100;
-                my $f   = $fract.floor;
-                $fract -= $f;
-                $fract-result ~= $f < 10 ?? "0$f" !!
-                                 (!$fract and $f %% 10) ?? ($f / 10).floor !! $f;
+            my $c;
+            if $!denominator < 100000 {
+                $c = 6;
+                $fract *= 1000000;
+            } else {
+                $c = $!denominator.chars + 1;
+                $fract *= nqp::pow_I(10, nqp::decont($c), Num, Int);
```

Comment:
The use of nqp ops seems a little inconsistent. Why use `nqp::pow_I` instead of just `**`? Or conversely, why use e.g., `$!denominator.chars` instead of `nqp::chars($!denominator.Str)` a line above?

Reply:
I did some testing both ways and the nqp::pow_I seemed to be marginally but measurably faster. I could easily change that back though. I am not really familiar with the different nqp ops and which give enough of a benefit to outweigh the added complexity. 

Thanks for your review.

Reply:
What you have is definitely not wrong, so now we're getting into personal preference. But since you're changing the algorithm used (and the original code is almost completely pure Perl 6), maybe a first commit that converts to the new algorithm, and then a second commit that changes some of the new code's slow parts to nqp ops?